### PR TITLE
User profile new

### PR DESCRIPTION
**Pull request by n16**
User profile is created and the upper section implemented. The middle of the screen is reserved for the event-list The sign in is connected to the user dashboard.

### CODE REVIEW

File: app/src/main/java/cmpe277/project/skibuddy/SignInActivity.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+
+                userProfileIntent.putExtra("UserProfileActivity", arrStr);
+//              SignInActivity.this.startActivity(userProfileIntent);
+                updateUI(true);
+                //TODO: end of to be deprecated
+
+                //instead of user profile, load dashboard after sign in
+                final Server s = new ServerSingleton().getServerInstance(this);
+                final String UID = "qwes";
```
Comment (ErnstHaagsman) [inline app/src/main/java/cmpe277/project/skibuddy/SignInActivity.java]: Could you change this to "G+" + Google+ profile ID number, it should be unique for each user. I just put "qwes" as a temporary thing for testing.

Thanks!

File: app/src/main/java/cmpe277/project/skibuddy/SignInActivity.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+                final String taglineStr = currentPerson.getTagline();
+                final String uriStr = currentPerson.getImage().getUrl();
+
+                //TODO: to be deprecated
+                arrStr.add(name);       //index 0
+                arrStr.add(taglineStr); //index 1
+                arrStr.add(uriStr);     //index 2
+
+
+                userProfileIntent.putExtra("UserProfileActivity", arrStr);
```
Comment (ErnstHaagsman) [inline app/src/main/java/cmpe277/project/skibuddy/SignInActivity.java]: Could you please remove the unused code?

File: app/build.gradle [hunk 3]
```diff
@@ -32,12 +32,18 @@ dependencies {
...
     compile 'com.android.support:appcompat-v7:23.0.1'
     compile 'com.android.support:design:23.1.0'
-    compile 'com.google.android.gms:play-services-plus:8.1.0'
+    //compile 'com.google.android.gms:play-services-plus:8.1.0'
 
     compile 'com.parse.bolts:bolts-android:1.+'
     compile 'com.parse:parse-android:1.+'
     compile project(':common')
-    compile 'com.google.android.gms:play-services-maps:8.1.0'
+    //compile 'com.google.android.gms:play-services-maps:8.1.0'
```
Comment (ErnstHaagsman) [inline app/build.gradle]: Could you please remove the commented out lines?

File: app/src/main/java/cmpe277/project/skibuddy/UserProfileActivity.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    ImageView imageView;
+
+    ArrayList<String> arrStr = new ArrayList<String>();
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.user_profile);
+
+        arrStr = getIntent().getStringArrayListExtra("UserProfileActivity");
```
Comment (ErnstHaagsman) [inline app/src/main/java/cmpe277/project/skibuddy/UserProfileActivity.java]: Get the user ID from the intent, and then load that user from the server

File: app/src/main/java/cmpe277/project/skibuddy/UserProfileActivity.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                            + PROFILE_PIC_SIZE;
+            new LoadProfileImage(imageView).execute(shorter_url);
+        }
+        else imageView.setImageBitmap(null);
+    }
+
+    /**
+     * Background Async task to load user profile picture from url
+     */
+    private class LoadProfileImage extends AsyncTask<String, Void, Bitmap> {
```
Comment (ErnstHaagsman) [inline app/src/main/java/cmpe277/project/skibuddy/UserProfileActivity.java]: I'm assuming that @AkankshaNagpal and @darshlakhani will need to load profile images as well, could you maybe refactor this class into a public class which takes a User object, and has an abstract onPostExecute? Then you could both use the same class.

**Comment by ErnstHaagsman** [comment]
The code looks pretty good, I'll pull it and run it later tonight. Please do address the comments that I posted in the code.

**Comment by ErnstHaagsman** [comment]
Also, if you're creating a new login, delete the previous 'MainActivity' class please.

File: app/src/main/java/cmpe277/project/skibuddy/UserProfileActivity.java (hunk 4 above)
Comment (n16) [inline app/src/main/java/cmpe277/project/skibuddy/UserProfileActivity.java]: The intent does not have uuid. It only has username, tagline, and imageStr

Comment (ErnstHaagsman) [inline app/src/main/java/cmpe277/project/skibuddy/UserProfileActivity.java]: The one you're creating doesn't have it, but the one @AkankshaNagpal creates will only hold the UUID. Use the key from BundleKeys to get it. 

```
     Bundle bundle = getIntent().getExtras();
     if (bundle != null) {
        String userUUID = bundle.getString(BundleKeys.UUID_KEY);
        s.getUser(UUID.fromString(userUUID), new ServerCallback 
```

etc.

Comment (n16) [inline app/src/main/java/cmpe277/project/skibuddy/UserProfileActivity.java]: bundle return null

**Comment by ErnstHaagsman** [comment]
We fixed the merge conflicts in this branch, and rebased it on Master in PR #28. I'll close this PR, but I'm merging your code (in #28)

**Pull request closed without merging**